lib/middleman-images/extension.rb

```diff
@@ -66,7 +69,16 @@ def add_processed_resource(source, process_options)
 
       def build_processed_path(source, process_options)
         destination = source.normalized_path.sub(/#{source.ext}$/, "")
-        destination += "-" + CGI.escape(process_options[:resize].to_s) if process_options[:resize]
+
+        options_part = process_options.slice(*Middleman::Images::Image::MINI_MAGICK_OPTIONS).keys
+                                                                                            .sort
+                                                                                            .map do |option_name|
+          CGI.escape(process_options[option_name].to_s.gsub(/\W+/, '_'))
+        end
+        options_part = options_part.join('-')
+        options_part = Digest::SHA1.hexdigest(options_part) if options_part.size > HASH_SIZE
+        destination += "-#{options_part}" if options_part.size > 0
+
         destination += "-opt" if process_options[:optimize]
```

Comment:
I think you could process the `optimize` option like any other option here. Or is there a reason to explicitly keep the "-opt" flag on the file name (@pmk1c ) ?

```suggestion
        options_part = process_options.slice(*OPTION_NAMES).keys
                                                                                            .sort
                                                                                            .map do |option_name|
          CGI.escape(process_options[option_name].to_s.gsub(/\W+/, '_'))
        end
        options_part = options_part.join('-')
        options_part = Digest::SHA1.hexdigest(options_part) if options_part.size > HASH_SIZE
        destination += "-#{options_part}" if options_part.size > 0
```

Reply:
In the case of the other options, the values will be added. So in the case of `resize: "400x225"` only the part `400x225` will be added to its name ([see also the tests for this](https://github.com/zweitag/middleman-images/blob/685778e848416b49aa3679bb020c1bef607a36ec/features/image_path.feature#L8)).

Thus, if we apply the same logic to the optimize part, only `true` would be added to the name. And IMHO `true` doesn't have any semantic value.

Reply:
Thank you, that was the part that was missing for me. Then I would suggest maybe adding the "-opt" string to the `options_part` string before hashing it. That would unify the output format.

Reply:
I just changed that